src/t_stream.c

```diff
@@ -2880,68 +2899,90 @@ void xpendingCommand(client *c) {
-
-        if (!strcasecmp(c->argv[3]->ptr, "IDLE")) {
-            if (getLongLongFromObjectOrReply(c, c->argv[4], &minidle, NULL) == C_ERR)
-                return;
-            if (c->argc < 8) {
-                /* If IDLE was provided we must have at least 'start end count' */
-                addReplyErrorObject(c,shared.syntaxerr);
-                return;
-            }
-            /* Search for rest of arguments after 'IDLE <idle>' */
-            startidx += 2;
-        }
-
-        /* count argument. */
-        if (getLongLongFromObjectOrReply(c,c->argv[startidx+2],&count,NULL) == C_ERR)
+    while(j < c->argc) {
+       if (!strcasecmp(c->argv[j]->ptr, "MKGROUP")) {
+          mkgroup = 1;
+	  j++;
+	  continue;
+       }
+       if (!strcasecmp(c->argv[j]->ptr, "IDLE")) {
+          if (j+1 == c->argc) {
+            addReplyErrorObject(c,shared.syntaxerr);
             return;
-        if (count < 0) count = 0;
-
-        /* start and end arguments. */
-        if (streamParseIntervalIDOrReply(c,c->argv[startidx],&startid,&startex,0) != C_OK)
+          }
+          if (getLongLongFromObjectOrReply(c, c->argv[j+1], &minidle, NULL) == C_ERR)
+            return;
+          if (c->argc < 8) {
+            /* If IDLE was provided we must have at least 'start end count' */
+            addReplyErrorObject(c,shared.syntaxerr);
+            return;
+          }
+          /* Search for rest of arguments after 'IDLE <idle>' */
+	  j = j+2;
+       }
+       else {
+	  if (j+2 >= c->argc) {
+            addReplyErrorObject(c,shared.syntaxerr);
+            return;
+          }
+	  /* count argument. */
+          if (getLongLongFromObjectOrReply(c,c->argv[j+2],&count,NULL) == C_ERR)
+            return;
+          if (count < 0) count = 0;
+            
+	  /* start and end arguments. */
+          if (streamParseIntervalIDOrReply(c,c->argv[j],&startid,&startex,0) != C_OK)
             return;
-        if (startex && streamIncrID(&startid) != C_OK) {
+          if (startex && streamIncrID(&startid) != C_OK) {
             addReplyError(c,"invalid start ID for the interval");
             return;
-        }
-        if (streamParseIntervalIDOrReply(c,c->argv[startidx+1],&endid,&endex,UINT64_MAX) != C_OK)
+          } 
+          if (streamParseIntervalIDOrReply(c,c->argv[j+1],&endid,&endex,UINT64_MAX) != C_OK)
             return;
-        if (endex && streamDecrID(&endid) != C_OK) {
+          if (endex && streamDecrID(&endid) != C_OK) {
             addReplyError(c,"invalid end ID for the interval");
             return;
-        }
+          }
 
-        if (startidx+3 < c->argc) {
-            /* 'consumer' was provided */
-            consumername = c->argv[startidx+3];
-        }
+          j = j + 3;
+          if(j < c->argc && strcasecmp(c->argv[j]->ptr, "MKGROUP")) {
+             /* 'consumer' was provided */
+             consumername = c->argv[j];
+             j++;
+	  }
+       }
     }
 
```

Comment:
Hi Oran, here I have one concern, but so far I have no better idea. Please comment it if you have better idea on it.
For the command xpending, it could be 
XPENDING <key> <group> [<start> <stop> <count> [<consumer>]]  
or
XPENDING <key> <group> [<start> <stop> <count> ] [MKGROUP]

The last argument could be <consumer> or [MKGROUP], so the only way to distinguish is to use strcasecmp(c->argv[j]->ptr, "MKGROUP").   If I move the MKGROUP augement to the front of the IDLE or <start>, it will break current change.
Thus, I choose my current code. 
If you have better arugment format or better coding logic, welcome to comment here.

Reply:
i see that in 6.2 we chose to extend this command by adding optional arguments (the IDLE) before the `<start>`.
so in this case, since MKGROUP isn't related to the range specification, i think it should come right after `<group>